visualization/src/pcl_visualizer.cpp

```diff
@@ -3611,7 +3642,12 @@ pcl::visualization::PCLVisualizer::renderViewTesselatedSphere (
   // * Compute area of the mesh
   //////////////////////////////
   vtkSmartPointer<vtkCellArray> cells = mapper->GetInput ()->GetPolys ();
-  vtkIdType npts = 0, *ptIds = nullptr;
+  vtkIdType npts = 0;
+#ifdef VTK_CELL_ARRAY_V2
```

Comment:
Please extract out this type. [Sample code](https://godbolt.org/z/G3-pRL)

Reply:
Is the point of extracting this out to eliminate multiple ifdefs everywhere its used?

I was searching for a good common place to define this type, but there doesn't appear to be any common header file.  Should it be (re)defined in each module where its necessary?

I'm assuming you mean something like:
```
#ifdef VTK_CELL_ARRAY_V2
using vtkCellPtsPtr = vtkIdType const *;
#else
using vtkCellPtsPtr = vtkIdType*;
```

Reply:
I am also looking into using iterators and it seems straightforward, but this change is still required

Reply:
Sorry, I seemed to have missed this conversation.

> Is the point of extracting this out to eliminate multiple ifdefs

:+1:

> Should it be (re)defined in each module where its necessary

That wouldn't be ideal. A single new header somewhere would be better, but I also don't know where.